mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/DynamicLoadProviderResourceDefinition.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.mod_cluster;
+
+import org.jboss.as.clustering.controller.ChildResourceDefinition;
+import org.jboss.as.clustering.controller.ReloadRequiredResourceRegistration;
+import org.jboss.as.clustering.controller.ResourceDescriptor;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.modcluster.load.impl.DynamicLoadBalanceFactorProvider;
+
+/**
+ * Definition for resource at address /subsystem=modcluster/proxy=X/dynamic-load-provider=configuration
+ *
+ * @author Radoslav Husar
+ */
+public class DynamicLoadProviderResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
+
+    // TODO this should be a resource at path load-provider=dynamic...
+    public static final PathElement PATH = PathElement.pathElement("dynamic-load-provider", "configuration");
```

Comment:
Should this TODO still be here?

Reply:
Actually, @rhusar, now would be the ideal time to make this change.

Reply:
@rhusar On second thought - let's do this in a separate PR.

Reply:
@jamezp Yes, they should still be there, TODOs I have added are to point out areas which are in my opinion problematic but beyond the scope of the RFE. Most of them are listed in the main issue's (https://issues.jboss.org/browse/WFLY-10439) linked "is followed up by" issues. Sadly, these were intended to be done timely in the same release so we don't have to do it in multiple models, however as you see this has been rotting here already for almost 2 months blocking progress on those....

Reply:
Okay that works. I wasn't sure if this was a reminder before commit TODO or a later TODO :)